tests/tensor_search/test_custom_vector_field.py

```diff
@@ -1051,7 +1051,460 @@ def test_lexical_search_with_custom_vector_field_searchable_attributes(self):
+            config=self.config, add_docs_params=AddDocsParams(
+                index_name=self.index_name_1,
+                docs=[{
+                    "_id": "0",
+                    "my_custom_vector": {
+                        "content": "custom content is here!!",
+                        "vector": self.random_vector_1
+                    }
+                }],
+                auto_refresh=True, device="cpu", mappings=self.mappings
+            )
+        )
+
+        # Confirm get_document_by_id returns correct content
+        res = tensor_search.get_document_by_id(
+            config=self.config, index_name=self.index_name_1,
+            document_id="0", show_vectors=True)
+        
+        # Check content is correct
+        assert res["_id"] == "0"
+        assert res["my_custom_vector"] == "custom content is here!!"
+
+        # Check tensor facets and embedding are correct
+        assert len(res[TensorField.tensor_facets]) == 1
+        assert res[TensorField.tensor_facets][0]["my_custom_vector"] == "custom content is here!!"
+        assert res[TensorField.tensor_facets][0][TensorField.embedding] == self.random_vector_1
+
+    def test_search_with_custom_vector_field(self):
+        """
+        Tensor search for the doc, with highlights
+        """
+        tensor_search.add_documents(
+            config=self.config, add_docs_params=AddDocsParams(
+                index_name=self.index_name_1,
+                docs=[
+                    {
+                        "_id": "custom_vector_doc",
+                        "my_custom_vector": {
+                            "content": "custom content is here!!",
+                            "vector": self.random_vector_1    # size is 512
+                        }
+                    },
+                    {
+                        "_id": "empty_content_custom_vector_doc",
+                        "my_custom_vector": {
+                            "vector": self.random_vector_2    # size is 512
+                        }
+                    },
+                ],
+                auto_refresh=True, device="cpu", mappings=self.mappings
+            )
+        )
+        
+        # Searching with context matching custom vector returns custom vector
+        # No text query given at all
+        res = tensor_search.search(
+            config=self.config, index_name=self.index_name_1,
+            search_method=SearchMethod.TENSOR,
+            context=SearchContext(**{"tensor": [{"vector": self.random_vector_1, "weight": 1}], }),
+        )
 
+        assert res["hits"][0]["_id"] == "custom_vector_doc"
+        assert res["hits"][0]["_score"] == 1.0
+        assert res["hits"][0]["_highlights"]["my_custom_vector"] == "custom content is here!!"
+
+    def test_search_no_query(self):
+        """
+        Tests that tensor search is possible with no text query, as long as context vector of correct length is given.
+        Vectorise should not be called at all.
+        """
```

Comment:
This is a good test - can this same test be applied to an index with a normal model?

Reply:
yep, this exact test is also in the test class above `test_search_no_query`, which uses a normal model :)